### ISSUE TITLE
Proposal: extending object-curly-spacing with rules for empty objects

### ISSUE THREAD

**Issue by LinusU** [eslint/eslint#7380]
**Tell us about your environment**
- **ESLint Version:** v3.8.0
- **Node Version:** v6.7.0
- **npm Version:** v3.10.7

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

```
{
  "rules": {
    "object-curly-spacing": ["error", "never"]
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

``` js
var a = { }
```

**What did you expect to happen?**

I expected that the rules should complain about the stray space between the curly braces `{ }`

I also think it would be a good idea to add a separate config for empty objects, since I think that people who have this rule set to `"always"` still doesn't want any whitespace inside empty objects.

**What actually happened? Please include the actual, raw output from ESLint.**

No error was raised

**Comment by kaicataldo** [comment]
This seems like a reasonable enhancement to me. As a user, I would expect "never" to warn about the case outlined above. Seems like this be added into the options object, perhaps as `emptyObject`.

With this option, the following would be valid:
```js
/* eslint object-curly-spacing: ["error", "never"]*/
var a = {foo:true};
var b = {};
```
```js
/* eslint object-curly-spacing: ["error", "never", { "emptyObject": true }] */
var a = {foo:true};
var b = { };
```
```js
/* eslint object-curly-spacing: ["error", "always", { "emptyObject": false }] */
var a = { foo:true };
var b = {};
```

One thing I'm not sure about is what we should do when the option is set to `"always"`. The current behavior is to just not enforce anything in the case of an empty object. I think it would be weird to enforce a space in empty objects by default - that's not a style I see recommended, and haven't really encountered that style myself.

```js
/* eslint object-curly-spacing: ["error", "always"] */
var a = { foo:true };
var b = { }; // What do we do with this case??
```

**Comment by LinusU** [comment]
Maybe `emptyObjects` should also be a string, `"always", "never"`, and default to `"never"`?

**Comment by platinumazure** [comment]
Agreed with @LinusU, we want to avoid boolean traps as much as we can.

**Comment by kaicataldo** [comment]
In general, I agree, but this rule has [other object options](http://eslint.org/docs/rules/object-curly-spacing#options) that use a boolean, and it feels like extra cognitive load for the user writing configuration to make this one different

**Comment by kaicataldo** [comment]
I'm not opposed to changing them all to `"always"` and `"never"`, though!

**Comment by kaicataldo** [comment]
Don't want to get tripped up on the names of the options - thoughts of the proposal (with the addition of changing the other option names in a backwards compatible way)?

**Comment by kaicataldo** [comment]
Closing this as discussion has stalled and it doesn't appear there is enough interest, either from the team or from the community, to continue the discussion at this time. While we wish we could accommodate all requests, we have limited resources and do need to prioritize. Please feel free to open a new issue with a proposal to continue this discussion if this is something you would like to continue advocating for.

**Comment by LinusU** [comment]
I would be happy to implement the rule (or addition to the `object-curly-spacing` rule), would that help? Who needs to "approve" this addition?

**Comment by kaicataldo** [comment]
We need a champion from the team who's willing to see the implementation through to its completion and three :+1:'s from the team.

The reason I closed this is that we've found that issues that get buried in the backlog without getting accepted end up getting lost (as can be evidenced by the lack of discussion here). Keeping our issues backlog manageable is necessary for keeping the project moving forward, both for us maintainers and also for our contributors.

Your being willing to help implement is definitely helpful! Feel free to open a new issue with your proposal if you'd like to get some discussion going around this - chances are much higher it'll get some attention when it's not multiple pages deep into our issues list.

**Comment by LinusU** [comment]
Sounds great, thank you for the thorough response 👍

**Comment by kaicataldo** [comment]
Anytime :)
